compiler/rustc_codegen_ssa/src/back/write.rs

```diff
@@ -943,7 +948,11 @@ fn execute_copy_from_cache_work_item<B: ExtraBackendMethods>(
             output_path.display()
         );
         match link_or_copy(&source_file, &output_path) {
-            Ok(_) => Some(output_path),
+            Ok(LinkOrCopy::Copy) => Some(output_path),
+            Ok(LinkOrCopy::Link) => {
```

Comment:
Is it actually important whether file was copied or linked here? Even if it was copied it still has the same content, so no reason to copy it back in the `copy_cgu_workproduct_to_incr_comp_cache_dir`.